proposal.rst

```diff
@@ -326,7 +326,7 @@ Use of Existing Code
 **If you already know about existing code you plan to incorporate or libraries
 you plan to use, please give details.**
 
-This project requires the use of existing code i.e. to import various open source file reading libraries such as Poppler, DjVuLibre and other available libraries from the Document Liberation Project.
+This project requires the use of existing code i.e. to import various open source file reading libraries such as Poppler, DjVuLibre and other available libraries from the Document Liberation Project which are free softwares under the GNU General Public License version 2 and later.
```

Comment:
If the `debian/copyright` file from the Debian packaging is accurate:

> Poppler is based on a Xpdf fork (updated at Xpdf 3.02), so the license
> of the poppler core library (libpoppler.so.X) is GPL v2 only.
> All the Poppler contributions are licensed as GPL v2+.

Which so the poppler core library is GPL v2, not GPL v2 and later (i.e. recipients of the code aren't able to upgrade their licence to a newer GPL version).

So your statement here seems to be inaccurate, but the situation (at least for Poppler - I didn't check the others) is not really a problem for what we want to do.

What we really need to be possible is for people to build omega with these libraries and distribute the resulting binaries - if that's not possible, then distros which ship binary packages (which is most of them) can't package the software.  Currently that's the case for our PHP bindings because the PHP licence is incompatible with the GPL licence, and that's annoying for users as they have to build the bindings themselves.  So what we actually need to check is that the libraries are under licences which are **compatible** with the GPL.

Reply:
1) Document Liberation project: "All participating libraries are released inter alia under the Mozilla Public License Version 2.0. The Document Liberation Project has selected this license to ensure that any new knowledge about a file format will be of benefit to all users"
https://www.gnu.org/licenses/license-list.en.html shows that Mozilla Public License Version 2.0 is compatible with the GPL v2.
2) DjVu : "DjVuLibre-3.5 was developed by Leon Bottou and others as a "Derived Work" of the DjVu Reference Library 3.5. As such, it is also subject to the GNU General Public License version 2."
3) From the above source, the license of the poppler core library is GPL v2.

The licenses of all the software libraries used are GPL v2 and thus compatible with the GPL. Am I correct?

Reply:
Yes, it looks like they are all compatible.